Utilities/File.cpp

```diff
@@ -1908,16 +1906,16 @@ const std::string& fs::get_config_dir()
 		std::string dir;
 
 #ifdef _WIN32
-		wchar_t buf[32768];
-		constexpr DWORD size = static_cast<DWORD>(std::size(buf));
-		if (GetEnvironmentVariable(L"RPCS3_CONFIG_DIR", buf, size) - 1 >= size - 1 &&
-			GetModuleFileName(nullptr, buf, size) - 1 >= size - 1)
+		constexpr DWORD size = 32768;
+		std::vector<wchar_t> buf(32768);
+		if (GetEnvironmentVariable(L"RPCS3_CONFIG_DIR", buf.data(), size) - 1 >= size - 1 &&
```

Comment:
It's not a lot of memory but what it should do instead of allocating always 32k is calling GetEnvironmentVariable in the start with one byte long buffer, the return value is the buffer size needed.

Reply:
done...